qCC/ccColorFromScalarDlg.cpp

```diff
@@ -142,19 +151,21 @@ ccColorFromScalarDlg::ccColorFromScalarDlg(QWidget* parent, ccPointCloud* pointC
 	for (unsigned i = 0; i < 4; i++)
 	{
 		m_colors[i] = ccColorScale::Shared(new ccColorScale(QString::asprintf("%d", i)));
+		updateChannel(i);
+		setDefaultSatValuePerChannel(i);
 	}
-	updateColormaps();
-
-	//update histograms
-	updateChannel(0);
-	updateChannel(1);
-	updateChannel(2);
-	updateChannel(3);
-	m_cloud->getCurrentDisplayedScalarField()->setColorScale(m_colors[3]); //set grey colour ramp to start with
 }
 
 ccColorFromScalarDlg::~ccColorFromScalarDlg()
 {
+	m_cloud->getCurrentDisplayedScalarField()->setColorScale(m_storedOrigColorScale);
+	m_cloud->getCurrentDisplayedScalarField()->setSaturationStart(m_storedOrigSatRange.min());
+	m_cloud->getCurrentDisplayedScalarField()->setSaturationStop(m_storedOrigSatRange.max());
+	m_cloud->getCurrentDisplayedScalarField()->setMinDisplayed(m_storedOrigDisplayRange.min());
+	m_cloud->getCurrentDisplayedScalarField()->setMaxDisplayed(m_storedOrigDisplayRange.max());
+	m_cloud->redrawDisplay();
```

Comment:
No need to call 'redrawDisplay' and 'prepareDisplayForRefresh + refreshDisplay'. The first one should be sufficient.